[FooNicholas] iP

FooNicholas opened the pull request:
# 📋 Yow Application Overview

- Simple and efficient task management chatbot
- Supports **ToDos**, *Deadlines*, and ~~Outdated Tasks~~
- User-friendly interface with command-line interactions

1. **Core Features**
   1. Add, mark, unmark, and delete tasks
   2. Search for tasks using keywords
   3. Supports deadlines and events with date-time formatting

```java
// Adding an Event Task Example
event Team Meeting /from 2024-10-01 1400 /to 2024-10-01 1600
 ```
- [x] Add ToDos
- [x] Add Deadlines
- [x] Add Events
- [ ] Implement recurring tasks
- [ ] Add priority tags

💡 Tip: Use the help command to view all available options.

"Productivity is never an accident. It is always the result of a commitment to excellence." 🚀

For more details, check the [Yow GitHub Repository](https://github.com/user/repo).

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

FooNicholas committed:
Implemented Rename, Greet, Exit

FooNicholas committed:
Implemented Echo

FooNicholas committed:
Implemented Add, List

FooNicholas committed:
Implemented Mark as Done

FooNicholas committed:
Implemented ToDo, Event, Deadline

FooNicholas committed:
Implemented Automated Text UI Testing

FooNicholas committed:
Implemented Exception Handling

FooNicholas committed:
Implemented Delete

FooNicholas committed:
Implement Level 7 and Javadoc Headers

FooNicholas committed:
Merge branch 'branch-Level-7'

Implement Save and Javadoc Headers

FooNicholas committed:
Implement Date and Time

FooNicholas committed:
Merge branch 'branch-Level-8'

Implement Date and Time

FooNicholas committed:
Implement MoreOOP

FooNicholas committed:
Implement Organize into Packages

FooNicholas committed:
Merge branch 'add-gradle-support'
Implement Gradle Support

FooNicholas committed:
Implement Gradle Usage

FooNicholas committed:
Implement JUnit tests for Deadlines and TaskList

FooNicholas committed:
Add JavaDoc comments

FooNicholas committed:
Merge branch 'branch-A-JavaDoc'

Add JavaDoc headers

FooNicholas committed:
Implement CS2103T Coding Standard

FooNicholas committed:
Merge branch 'A-CodingStandard'

Implement CS2103T Coding Standard

FooNicholas committed:
Implement Find

FooNicholas committed:
Merge branch 'branch-Level-9'

Implement Find

shawnnygoh commented on src/main/java/yow/Ui.java:
```diff
@@ -0,0 +1,37 @@
+        splitLine();
+    }
+
+    void helpChat() {
```
Perhaps helpChat() could be a verb instead?
```suggestion
    void showHelpChat() {
```

shawnnygoh commented on src/main/java/yow/Ui.java:
```diff
@@ -0,0 +1,37 @@
+    }
+
+
+    void startChat() {
```
Perhaps startChat() could be a verb instead?
```suggestion
    void showStartChat() {
```

shawnnygoh commented on src/main/java/yow/Ui.java:
```diff
@@ -0,0 +1,37 @@
+        prettyPrint("Hello! I'm Yow\nWhat can I do for you yow?");
+    }
+
+    void endChat() {
```
Perhaps endChat() could be a verb instead?
```suggestion
    void showEndChat() {
```

shawnnygoh reviewed (commented):
Looks good. 👍🏻
Code mostly follows coding standards.
Good job!

Masunori commented on src/main/java/yow/Deadlines.java:
```diff
@@ -0,0 +1,61 @@
+/**
+ * Represents a task with a deadline.
+ * A yow.Deadlines task has a description, completion status, and a due date.
+ */
```
Plural names may make people mistake this class for a collection. How about removing the 's' for all subclasses of Task?

Masunori commented on src/main/java/yow/Events.java:
```diff
@@ -0,0 +1,69 @@
+ */
+public class Events extends Task {
+    private LocalDateTime start;
+    private LocalDateTime end;
```
I like that you try to handle more than one format to give users some flexibility! 👍

Masunori commented on src/main/java/yow/Events.java:
```diff
@@ -0,0 +1,69 @@
+        } catch (DateTimeParseException e1) {
+            try {
+                return LocalDateTime.parse(date, INPUT_FORMATTER2);
+            } catch (DateTimeParseException e2) {
```
I love that you tell the user the correct way to do when the exception is thrown!

Masunori commented on src/main/java/yow/Events.java:
```diff
@@ -0,0 +1,69 @@
+     * Converts the Events task into a formatted string for file storage.
+     *
+     * @return A string representation of the task suitable for file storage.
+     */
```
Good use of Tell, Don't Ask!

Masunori commented on src/main/java/yow/Parser.java:
```diff
@@ -0,0 +1,161 @@
+    }
+
+    boolean parseCommand(String userInput) {
+
```
My approach for this is to replace the switch-case with a map to map from the command to the action:
```java
new HashMap<String, Function<Void, Void> dummyNameHashMap;
```

Then, you can do:
```java
try {
    this.dummyNameHashMap.get(userInput.split{" ")[0]).apply();
} catch (YowException e) {
    // something
}
```

But of course, that is just my design, and it does not mean that any design is better than the other!

Masunori commented on src/main/java/yow/Parser.java:
```diff
@@ -0,0 +1,161 @@
+        }
+        return trim;
+    }
+
```
Great naming convention! It clearly tells me that these methods have "the same type".

Masunori commented on src/main/java/yow/Parser.java:
```diff
@@ -0,0 +1,161 @@
+                case "bye":
+                    return false;
+                case "list":
+                    ui.prettyPrint(taskList.stringifyList());
```
How about encapsulating this in a `handleListCommand` to keep it consistent with other cases?

Masunori commented on src/main/java/yow/Task.java:
```diff
@@ -0,0 +1,54 @@
+/**
+ * Represents an abstract task with a description and completion status.
+ * This serves as a base class for different types of tasks such as ToDos, yow.Deadlines, and Events.
+ */
```
Very structured! 👍

Masunori commented on src/main/java/yow/Parser.java:
```diff
@@ -0,0 +1,161 @@
+        taskList.addTask(event);
+        ui.prettyPrint("Got it yow. I've added this task:\n  " + event);
+    }
+
```
Have you handled the case when the index is outside the list of task range?

Masunori commented on src/test/java/main/yow/DeadlinesTest.java:
```diff
@@ -0,0 +1,31 @@
+import yow.YowException;
+
+public class DeadlinesTest {
+
```
A little nitpicking here. The naming convention for test methods is:

```java
public void testFeature_testScenario_expectedBehaviour();
```

For example:

```java
public void dateParse_deadlineTask_correctParsing();
```

FooNicholas committed:
Implement GUI for Yow

FooNicholas committed:
Implement GUI for Yow

FooNicholas committed:
Add assertions to enforce input validation

Added assertions to enforce input validation in key methods:
- Ensured non-null values in Storage, TaskList, and Parser.
- Checked valid index ranges before accessing task lists.
- Validated proper command formatting before parsing.

These assertions help detect unexpected inputs early, improving
code robustness without affecting production performance
(since assertions can be disabled).

FooNicholas committed:
Improve main method documentation in Yow.java

Refined the method header comment for the main method in Yow.java:
- Clarified the method’s role as the entry point for the chatbot.
- Provided a more precise overview of its initialization steps.
- Added missing exception documentation for YowException.

This improves code readability and ensures that the documentation
adheres to the coding standard for method headers.

FooNicholas committed:
Merge pull request #2 from FooNicholas/branch-A-Assertions

Add assertions to enforce input validation

FooNicholas committed:
Merge branch 'master' into branch-A-CodeQuality
Merge master into A-CodeQuality

FooNicholas committed:
Merge pull request #3 from FooNicholas/branch-A-CodeQuality

PR for Branch-A-CodeQuality

FooNicholas committed:
Add Do Within a Period Task

FooNicholas committed:
Merge pull request #4 from FooNicholas/branch-BCD-Extension

Add Do Within a Period Task

FooNicholas committed:
Refactor file names and styled GUI elements

FooNicholas committed:
Refactor README and minor fixes

FooNicholas committed:
Add updated README
